### ISSUE TITLE
Replicate METviewer performance diagram

### ISSUE THREAD

**Issue by bikegeek** [dtcenter/METplotpy#20]
## Describe the New Feature ##
Create a performance diagram plot to replace the R implementation currently in use by METviewer.  In addition to supporting METviewer, this plot should be invocable as a stand-alone script.  Refer to the performance diagram on the METplotpy/calcpy requirements gathering doc:
https://docs.google.com/document/d/1KUHgpB-hDqdDOa83mpoQn9SMl2BXwL4PrVXvqG4xayg/edit#

### Acceptance Testing ###
Database view from the database employed by METviewer:
- use data supplied by Tatiana Burek: 

[plot_20200317_151252_data.txt](https://github.com/NCAR/METplotpy/files/4411593/plot_20200317_151252_data.txt)

Visually compare the plot created with the Python implementation against the one created via METviewer to verify that lines and time series are consistent.  Note, the look-and-feel of the implementations may differ as R and matplotlib implementations will vary. 

### Time Estimate ###
- [X] 3 days: Reverse engineer John David Gagne and others'  implementation of the performance diagram axes, equal lines of Bias and equal lines of CSI (3 days)
- [X] 3 days: Create a simple, hard-coded performance diagram on "dummy" data and finalize markers, colors, font sizes, legend locations (4 days)
- [x] 3 days: Incorporate support for plot settings in YAML configuration files-based on settings that are required for compatibility with METviewer
-[ x] ~1 day verify/test that the default config values are used in the absence of custom config file values
- [x] ~3 days: Reverse engineer R script currently in use by METviewer to replicate the creating of time series points from input data.  Design Python version and submit for design review before generating any Python code.
- [x] ~1 day: Implement support for ordering series plotting attributes based on the series_ordering setting in the config file
 -[x] ~2 days: Implement support for creating time series data for any type of input data
 -[x] 1 day: Implement support for plotting confidence intervals/error bars for PODY
- [x] 1 day: Provide support for invoking the performance diagram script in METviewer, where a config file is included in the invocation (ie python performance_diagram.py <config filename>)
- [x] 4 days: Incorporate Python implementation into METviewer (anticipating that this will be iterative to address anything that is missing/mismatched)
- [x] 1 day: Create test that utilizes METcalcpy image comparison.
- [x] 2 days: Provide support for setting xtlab_size, ytlab_size (requested by Ho Chun at EMC)
- [x] 2 days: Provide support for setting xlab_size, and ylab_size (requested by Ho Chun at EMC) 
- [x] 2 days: Provide support for setting caption
- [X] 2 days: Provide support for setting legend box location

### Relevant Deadlines ###
alpha implementation should be complete before June 2020:

- configurable elements are supported via a default and/or custom YAML configuration file

- Python implementation in use by METviewer (replacing R implementation)

### Funding Source ###
27700041 - up to 20 hours 
2760831 - up to 10 hours
2760821 for the rest

### Assignee ###
- [x] Select Minna Win, Hank Fisher
- [x] Select **scientist Tara Jensen

### Labels ###
- [x] Select **component(s)**  plot stat data
- [x] Select **priority** high
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Review **projects** and select relevant **Repository** and **Organization** ones
- [x] Select **milestone**

## Define Related Issue(s) ##
Consider the impact to other METplus components.
Delete items that do not apply.

- [x] Define new issue for METcalcpy.  Github Issue #7

**Comment by bikegeek** [comment]
Unsupported:
==========
xtlab_decim, xtlab_horiz, xtlab_perp, ytlab_decim, ytlab_horiz, and ytlab_perp will NOT be supported as there is no way to support this in Plotly and we want plots to behave as consistently as possible. 

Supported:
=========
caption:
-color
-font size
-limited font styling:normal, italic, and oblique 

legend:
-color
-up/down location with respect to x-axis
-left/right location with respect to y-axis
-number of columns
-surrounding box or no surrounding box

x-label:
-font size
-tick label size
-tick label orientation

y-label:
-font size
-tick label size
-tick label orientation
